### ISSUE TITLE
SlicerVMTK extension

### ISSUE THREAD

**Issue by spujol** [Slicer/Slicer#5187]
The SlicerVMTK extension appears as 'Vascular Modeling Toolkit' in the list of Slicer modules. A search for 'Slicer VMTK' in the module finder doesn't give any result.

**Comment by lassoan** [comment]
Extension name is often not the same as module category name (extensions often add modules to existing categories).

To show only modules that are provided by extensions, uncheck "Built-in" category in module finder:

![image](https://user-images.githubusercontent.com/307929/93411696-bc24bc00-f869-11ea-92fa-4fef9fd05696.png)

**Comment by spujol** [comment]
Could the search functionality include the module description? In that example, 'SlicerVMTK' can be found in the Location section.

**Comment by lassoan** [comment]
Yes. If you check "Full text" then module description is included in the search, too.

**Comment by spujol** [comment]
The search for 'SlicerVMTK' with the option 'Full text' checked and 'Built-in' unchecked did not produce any result.
(Slicer4.11.0-2020-09-15 on MacOSX 10.14.16)

**Comment by spujol** [comment]
The search for 'VMTK' returned the Centerline Metrics module only

**Comment by lassoan** [comment]
Module name, description, authors, etc. are search in "full text" mode. If VMTK is not mentioned then the module will not be found.

Unchecking built-in should make it easy to find all modules that a user installed using extensions. What is your exact use case?

Extensions are only used for installing modules, after that the modules are not aware where they come from (other than their install location), so there would be no clean and easy way of figuring out if a module is provided by a specific extension. Consistently placing specific words in the module description could help further filtering of non-built-in modules.

**Comment by spujol** [comment]
Since the extension name is included in the install location, could the the path be included in the search? 
'SlicerVMTK' appears in the path of the Location section in all four modules (Centerline metrics, Extract Centerline, Level Set Segmentation and Vesselness Filtering).

**Comment by lassoan** [comment]
> could the path be included in the search?

We would need to filter out other parts of the path (username, potentially project subfolder, etc. and standard installation subdirectories). Since the path can be anything (may be provided by the user) and its structure depends on the operating-system, this filtering is not that trivial.

Do you find it difficult to find all the modules that you need by simply checking "built-in"?

**Comment by jamesobutler** [comment]
Maybe a better category name would be "VMTK" if that is the more common usage than "Vascular Modeling Toolkit" and put the full name in the description text.

@lassoan Could you support filtering by picking from a known list of categories?  Category navigation was the highest level navigation in the older Slicer modules navigation, so being able to support getting the list of all modules under the say "IGT" category would be helpful. It's difficult to know what to search for if you don't know the list of available category names.

**Comment by lassoan** [comment]
Browsing by category is still available (via the menu, as before), but it could be added to full-text search, too.

**Comment by spujol** [comment]
That would be great
